config/clients/java/template/OpenFgaApiTest.java.mustache

```diff
@@ -0,0 +1,709 @@
+
+        // When
+        CreateStoreResponse response = fga.createStore(new CreateStoreRequest());
+
+        // Then
+        mockHttpClient.verify().post("https://localhost/stores").called(1);
+        assertEquals(DEFAULT_STORE_ID, response.getId());
+        assertEquals(DEFAULT_STORE_NAME, response.getName());
+    }
+
+    @Test
+    public void createStore_bodyRequired() throws ApiException {
+        // When
+        ApiException exception = assertThrows(ApiException.class, () -> fga.createStore(null));
+
+        // Then
+        assertEquals("Missing the required parameter 'body' when calling createStore", exception.getMessage());
+    }
+
+    /**
+     * Delete a store.
+     */
+    @Test
+    public void deleteStoreTest() throws ApiException {
+        // Given
+        mockHttpClient
+                .onDelete("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X")
+                .doReturn(200, "");
+
+        // When
+        fga.deleteStore(DEFAULT_STORE_ID);
+
+        // Then
+        mockHttpClient
+                .verify()
+                .delete("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X")
+                .called(1);
+    }
+
+    @Test
+    public void deleteStore_storeIdRequired() throws ApiException {
+        // When
+        ApiException exception = assertThrows(ApiException.class, () -> fga.deleteStore(null));
+
+        // Then
+        assertEquals("Missing the required parameter 'storeId' when calling deleteStore", exception.getMessage());
+    }
+
+    /**
+     * Expand all relationships in userset tree format, and following userset rewrite rules.  Useful to reason about and debug a certain relationship.
+     */
+    @Test
+    public void expandTest() throws ApiException {
+        // Given
+        String expectedBody = String.format(
+                "{\"tuple_key\":{\"object\":\"%s\",\"relation\":\"%s\",\"user\":\"%s\"},\"authorization_model_id\":\"01G5JAVJ41T49E9TT3SKVS7X1J\"}",
+                DEFAULT_OBJECT, DEFAULT_RELATION, DEFAULT_USER);
+        mockHttpClient
+                .onPost("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X/expand")
+                .withBody(is(expectedBody))
+                .doReturn(200, "{}");
+        ExpandRequest request = new ExpandRequest()
+                .authorizationModelId(DEFAULT_AUTH_MODEL_ID)
+                .tupleKey(new TupleKey()
+                        ._object(DEFAULT_OBJECT)
+                        .relation(DEFAULT_RELATION)
+                        .user(DEFAULT_USER));
+
+        // When
+        ExpandResponse response = fga.expand(DEFAULT_STORE_ID, request);
+
+        // Then
+        mockHttpClient
+                .verify()
+                .post("https://localhost/stores/01YCP46JKYM8FJCQ37NMBYHE5X/expand")
+                .withBody(is(expectedBody))
+                .called(1);
+        assertNull(response.getTree());
```

Comment:
do we have a test asserting when a non-null response given?

Reply:
I should add a tree to this response

Reply:
Done in [9be1e1f](https://github.com/openfga/sdk-generator/pull/170/commits/9be1e1fec5f2905f317b77ee253f4c250f3e1b1d)